src/command_line_parser.cc

```diff
@@ -1912,8 +1923,21 @@ CLParser::ParseCommandLine(int argc, char** argv)
   }
 
   // In fixed schedule mode, the request count is based on the number of
-  // payloads in the user-provided dataset
+  // payloads in the user-provided dataset (minus warmup request count)
   if (params_->inference_load_mode == pa::InferenceLoadMode::FixedSchedule) {
+    if (params_->request_count != 0) {
+      throw std::runtime_error(
+          "Cannot specify `--request-count` when using `--fixed-schedule`. The "
+          "request count will be equal to the number of requests in the input "
+          "data minus the warmup request count.");
+    }
+
+    if (params_->user_data.empty()) {
+      throw std::runtime_error(
+          "When using `--fixed-schedule`, you must provide a dataset using "
+          "`--input-data`.");
+    }
+
```

Comment:
As a bit of a wrinkle when adding warmup support for fixed schedule, the number of requests in the input data JSON given to PA needs to be known quite early in the application flow.

This code quickly reads the JSON to get the number of entries, which is then used for validation purposes and later for proper thread allocation.

This wasn't my ideal solution, but it was the best middle ground I could come up with without severely redesigning other parts of PA.